Posix/Win32 main refactor

yol opened the pull request:
Follow-up from #12207 

## Description
Remove TARGET_POSIX ifdefs from posix/main.cpp and make win32 platform independent of posix/main.cpp

## Motivation and Context
WinMain() currently uses posix/main.cpp, which is unexpected and unnecessary.

## How Has This Been Tested?
Linux x86_64 works, other builds could fail - will update according to Jenkins results

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

pkerling committed:
Remove superfluous #ifdefs and reorganize includes accordingly

This code is specific to POSIX and in a .cpp file, so the #ifdefs
TARGET_POSIX and __cplusplus are unwarranted.

pkerling committed:
Fix CAppParamParser::Parse argument type const-ness

A const char*[] is basically a two-dimensional array that only has
const-ness on one level, which means that the contents are not really
const. Replace with const char* const* to have both levels const.
This also removes the need for some const_cast hacks in the startup
code.

pkerling committed:
Remove hackish const_casts

pkerling committed:
Correctly set DEBUG log level for debug builds

The previous code setting LOG_LEVEL_DEBUG did nothing since
CAdvancedSettings::Initialize() sets the level to LOG_LEVEL_NORMAL
anyway. Fix this by overriding the level after possible initialization.

pkerling committed:
Remove log level setting from posix main()

The initial log level is LOG_LEVEL_DEBUG anyway, and setting
LOG_LEVEL_NORMAL with handling the _DEBUG #ifdef override is now handled
in XBMC_Run.

pkerling committed:
Do not use posix main() in WinMain

since it is very confusing for the win32 platform to include code from
platform/posix. It also only saves a few lines.

pkerling committed:
Remove renderGUI var from posix main()

Was never used anyway.

pkerling committed:
Fix posix main() debug detection ifdef from DEBUG to _DEBUG

_DEBUG is set by CMake for debug builds, DEBUG is not.

pkerling committed:
Make WinMain.cpp the CORE_MAIN_SOURCE for the win32 platform

Otherwise, posix/main.cpp would still be used

pkerling committed:
Fix OS X build

HAS_SDL is defined by system.h, so it must be included before the #ifdef

pkerling commented:
jenkins build this please

MartijnKaijser commented:
Only authorized people can trigger jenkins manually ;) 
You can do this by simply repushing to a branch.

pkerling commented:
It did trigger the build though? Or was that coincidence?
And I can't repush since nothing has changed. Jenkins just failed the build due to internal issues previously.

fritsch commented:
jenkins is happy - so all fine here.

FernetMenta reviewed (approved):
looks ok to me

Rechi commented:
In #11080 set log level to debug logging for debug builds was disabled, so I'm not sure if we should change it back again.

MartijnKaijser commented:
Indeed debuglogging shouldn't be auto enabled.

pkerling commented:
Ah didn't know about #11080 - I just assumed that forcing DEBUG was the original intention of the broken code. But seems it was just leftovers. Removing the #ifdef _DEBUG in xbmc.cpp should fix it then.

Rechi commented:
This was just forgotten in #11080 because it didn't wok as expected.

pkerling committed:
Remove default log setting of DEBUG in debug builds

Turns out that this code was actually just overlooked in an earlier cleanup because it did not work and is not actually supposed to be here anymore.

pkerling commented:
Related code removed from xbmc.cpp

MartijnKaijser commented:
No objections?
